include/ensmallen_bits/nsga2/nsga2_impl.hpp

```diff
@@ -0,0 +1,311 @@
+  std::map<size_t, std::set<size_t> > dominated;
+
+  // reset and intialize fronts
+  fronts.clear();
+  fronts.push_back(std::vector<size_t>());
+
+  for (size_t p = 0; p < populationSize; p++) {
+    dominated[p] = std::set<size_t>();
+    dominationCount[p] = 0;
+
+    for(size_t q=0; q < populationSize; q++) {
+      if (Dominates(calculatedObjectives, p, q)) {
+        dominated[p].insert(q);
+      }
+      else if (Dominates(calculatedObjectives, q, p)) {
+        dominationCount[p] += 1;
+      }
+    }
+
+    if (dominationCount[p] == 0) {
+      ranks[p] = 0;
+      fronts[0].push_back(p);
+    }
+  }
+
+  size_t i = 0;
+
+  while (fronts[i].size() > 0) {
+    std::vector<size_t> nextFront;
+
+    for (size_t p: fronts[i]) {
+      for (size_t q: dominated[p]) {
+        dominationCount[q]--;
+
+        if (dominationCount[q] == 0) {
+          ranks[q] = i + 1;
+          nextFront.push_back(q);
+        }
+      }
+    }
+
+    i++;
+    fronts.push_back(nextFront);
+  }
+}
+
+inline bool NSGA2::Dominates(std::vector<arma::vec> calculatedObjectives,
+                             size_t candidateP,
+                             size_t candidateQ)
+{
+  Info << "NSGA2::NSGA2::Dominates()" << std::endl;
+
+  bool all_better_or_equal = true;
+  bool atleast_one_better = false;
+  size_t n_objectives = calculatedObjectives[0].n_elem;
+
+  for (size_t i = 0; i < n_objectives; i++) {
+    if (calculatedObjectives[candidateP](i) > calculatedObjectives[candidateQ](i)) {
+      // p.i is worse than q.i for the i-th objective function
+      all_better_or_equal = false;
+    }
+
+    if (calculatedObjectives[candidateP](i) < calculatedObjectives[candidateQ](i)) {
+      // p.i is better than q.i for the i-th objective function
+      atleast_one_better = true;
+    }
+  }
+
+  return all_better_or_equal and atleast_one_better;
+}
+
+template<typename MultiobjectiveFunctionType>
+inline void NSGA2::CrowdingDistanceAssignment(std::vector<size_t> front,
+                                       MultiobjectiveFunctionType objectives,
+                                       std::vector<double>& crowdingDistance)
+{
+  if (front.size() > 0) {
+    for(size_t elem: front) {
+      crowdingDistance[elem] = 0;
+    }
+
+    size_t fSize = front.size();
+
+    for(size_t m = 0; m < objectives.NumObjectives(); m++) {
+      crowdingDistance[front[0]] = objectives.GetMaximum(m);
+      crowdingDistance[front[fSize - 1]] = objectives.GetMaximum(m);
+
+      for(size_t i = 1; i < fSize - 1 ; i++) {
+        crowdingDistance[front[i]] += (crowdingDistance[front[i-1]] - crowdingDistance[front[i+1]])/(objectives.GetMaximum(m) - objectives.GetMinimum(m));
+      }
+    }
+  }
+}
+
+inline bool NSGA2::CrowdingOperator(size_t idxP,
+                             size_t idxQ,
+                             std::vector<size_t> ranks,
+                             std::vector<double> crowdingDistance)
+{
+  if (ranks[idxP] < ranks[idxQ]) {
```

Comment:
Please put your braces in a new line, as per the [brace placement style guideline](https://github.com/mlpack/mlpack/wiki/DesignGuidelines#brace-placement). I think there are multiple places you can do this.